rotkehlchen/api/rest.py

```diff
@@ -4675,15 +4680,48 @@ def export_history_events(
             )
 
         if len(history_events) == 0:
-            return api_response(
-                wrap_in_fail_result('No history processed in order to perform an export'),
+            return wrap_in_fail_result(
+                message='No history processed in order to perform an export',
                 status_code=HTTPStatus.CONFLICT,
             )
 
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            settings = self.rotkehlchen.get_settings(cursor)
+            currency = settings.main_currency.resolve_to_asset_with_oracles()
+
         serialized_history_events = []
         headers: dict[str, None] = {}
+        if currency == A_USD:  # if the currency is not USD don't loose time querying prices that won't be used  # noqa: E501
+            entries = dbevents.get_base_entries_missing_prices(EvmEventFilterQuery.make())
+            query_missing_prices_of_base_entries(
+                database=self.rotkehlchen.data.db,
+                entries_missing_prices=entries,
+            )
+
         for event in history_events:
-            serialized_event = event.serialize_for_csv()
+            if currency != A_USD:
```

Comment:
keep it simple. Remove the above query_missing_prices_of_base_entries and if it's A_USD and `usd_value` is zero also continue with the query here